ulib/experimental/Steel.Memory.fst

```diff
@@ -54,6 +62,13 @@ let disjoint_addr (m0 m1:heap) (a:addr)
       summable_permissions p0 p1 /\
       t0 == t1 /\
       v0 == v1
+    | Some (Array t0 len0 seq0), Some (Array t1 len1 seq1) ->
+      t0 == t1 /\
+      len0 == len1 /\
+      (forall (i:nat{i < len0}).
+        let (x0, p0) = Seq.index seq0 i in
+	let (x1, p1) = Seq.index seq1 i in
+	x0 == x1 /\ summable_permissions p0 p1)
```

Comment:
In order to support array splitting, I was expecting the definition of disjoint_addr in this case to be something like:

```
    | Some (Array t0 len0 s0), Some (Array t1 len1 s1) ->
      t0 == t1 /\
      len0 == len1 /\
      (forall i.
         let v0, p0 = s0.[i] in
         let v1, p1 = s1.[i] in
         p0 +. p1 <= 1.0R /\
         (0.0R < p0 && 0.0R < p1 ==> v0 == v1))
```

i.e., the array contents agree on those cells where both sides have non-zero permission

Reply:
So right now the permission in each cell has to be `> 0`. But indeed, if we want to support splitting, then we need to allow some cells to be `0`. Then, the invariant on a array cell should be that at least one of the cells has non-zero permission.

Reply:
I'll have to change that as well as the `disjoint_addr` definition for arrays as you noted.

Reply:
The predicate has been changed